common/oatbox/cache/GcpTokenCacheItemPool.php

```diff
@@ -0,0 +1,251 @@
+
+        $this->log(__METHOD__, $keys);
+
+        $items = $this->getCache()->getItems($keys);
+
+        foreach ($items as $item) {
+            $this->unSerializeIfNecessary($item);
+        }
+
+        return $items;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function hasItem($key)
+    {
+        $key = $this->getCacheKey();
+
+        $this->log(__METHOD__, [$key]);
+
+        return $this->getCache()->hasItem($key);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function clear()
+    {
+        $this->log(__METHOD__);
+
+        return $this->getCache()->clear();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function deleteItem($key)
+    {
+        $key = $this->getCacheKey();
+
+        $this->log(__METHOD__, [$key]);
+
+        return $this->getCache()->deleteItem($key);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function deleteItems(array $keys)
+    {
+        $keys = [$this->getCacheKey()];
+
+        $this->log(__METHOD__, $keys);
+
+        return $this->getCache()->deleteItems($keys);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function save(CacheItemInterface $item)
+    {
+        $this->log(__METHOD__);
+
+        if ($this->isWritingDisabled()) {
+            return true;
+        }
+
+        $this->serializeToPersist($item);
+
+        return $this->getCache()->save($item);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function saveDeferred(CacheItemInterface $item)
+    {
+        $this->log(__METHOD__);
+
+        if ($this->isWritingDisabled()) {
+            return true;
+        }
+
+        $this->serializeToPersist($item);
+
+        return $this->getCache()->saveDeferred($item);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function commit()
+    {
+        $this->log(__METHOD__);
+
+        if ($this->isWritingDisabled()) {
+            return true;
+        }
+
+        return $this->getCache()->commit();
+    }
+
+    private function log(string $method, array $keys = []): void
+    {
+        if ($this->isDebug()) {
+            $this->getLogger()->debug(
+                sprintf(
+                    '[%s] Called method %s with key(s) %s',
+                    __CLASS__,
+                    $method,
+                    implode(',', $keys)
+                )
+            );
+        }
+    }
+
+    private function isDebug(): bool
+    {
+        if ($this->isDebug === null) {
+            $this->isDebug = (bool)$this->getOption(self::OPTION_ENABLE_DEBUG, false);
+        }
+
+        return $this->isDebug;
+    }
+
+    private function isWritingDisabled(): bool
+    {
+        if ($this->isWritingDisabled === null) {
+            $this->isWritingDisabled = (bool)$this->getOption(self::OPTION_DISABLE_WRITE, false);
+        }
+
+        return $this->isWritingDisabled;
+    }
+
+    private function getCache(): CacheItemPoolInterface
+    {
+        $this->cache = $this->createCache();
```

Comment:
Do u think we should check if this attribute has a value before to avoid to recreate it if it already exists?

Reply:
Good point, I will had this and removed after refactoring :)